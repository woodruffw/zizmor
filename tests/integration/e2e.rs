//! End-to-end snapshot integration tests.

use anyhow::Result;

use crate::common::{input_under_test, zizmor, OutputMode};

#[cfg_attr(not(feature = "gh-token-tests"), ignore)]
#[test]
fn gha_hazmat() -> Result<()> {
    // Stability test against with online retrieval but no online audits.
    // Ensures that we consistently collect the same files in the default
    // configuration.
    insta::assert_snapshot!(zizmor()
        .offline(false)
        .output(OutputMode::Both)
        .args(["--no-online-audits"])
        .input("woodruffw/gha-hazmat@42064a9533f401a493c3599e56f144918f8eacfd")
        .run()?);
    Ok(())
}

#[test]
fn menagerie() -> Result<()> {
    insta::assert_snapshot!(zizmor()
        .output(OutputMode::Both)
        .input(input_under_test("e2e-menagerie"))
        .run()?);

    Ok(())
}
