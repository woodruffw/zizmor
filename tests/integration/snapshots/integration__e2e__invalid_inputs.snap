---
source: tests/integration/e2e.rs
expression: "zizmor().expects_failure(true).input(input_under_test(&format!(\"invalid/{tc}.yml\"))).run()?"
---
fatal: no audit was performed
failed to register input: @@INPUT@@

Caused by:
    0: invalid input structure
       
       Caused by:
           0: invalid GitHub Actions workflow: file://@@INPUT@@
           1: jobs.invalid: "runs-on" is a required property
              jobs.invalid: Additional properties are not allowed ('steps' was unexpected)
              jobs.invalid: "uses" is a required property
              
    1: invalid GitHub Actions definition: file://@@INPUT@@
       
       Caused by:
           Additional properties are not allowed ('jobs', 'on', 'permissions' were unexpected)
           "description" is a required property
           "runs" is a required property
           
    2: failed to register input as workflow or action
